### ISSUE TITLE
Deserialization of segwit blocks fails on version 0.8 with Bitcoin Core 0.15.0RC2

### ISSUE THREAD

**Issue by harding** [petertodd/python-bitcoinlib#146]
Using Bitcoin Core 0.15.0RC2 and python-bitcoinlib 0.8 installed via pip, I receive the error `bitcoin.core.serialize.SerializationError: Asked to read 0x4c8f597226cf3302 bytes; MAX_SIZE exceeded`

**Steps to reproduce**

1. On Debian 8, Install Bitcoin Core 0.15.0RC2 from bitcoin.org/bin .  Start it and sync the block chain (note: I synced on previous releases)
2. `virtualenv -p python3`
3. `. venv/bin/activate`
4. `pip install python-bitcoinlib`
5. Create and run the following script.

```python3
import bitcoin
from bitcoin import rpc

proxy = rpc.Proxy()

def print_block(height):
    print(proxy.getblock(proxy.getblockhash(height)))

print("Version:", bitcoin.__version__, "\n")

print("This works for me")
print_block(480000)

print("\nThis fails for me")
print_block(482439)
```

**Results**

```text
(venv)harding@redspot:~/ref-payment-batching$ python3 fail-case.py 
Version: 0.8.0 

This works for me
CBlock(536870914, lx(00000000000000000095e53a3a2707c219d7e00e046f320b64f6c41431a0fdab), lx(0f2a2069d19d8e0ca298fe452fd0213be5035dff388868d6bebb3905d0c90c70), 1502407559, 0x180130e0, 0xa2f2df87)

This fails for me
Traceback (most recent call last):
  File "fail-case.py", line 16, in <module>
    print_block(482439)
  File "fail-case.py", line 7, in print_block
    print(proxy.getblock(proxy.getblockhash(height)))
  File "/home/harding/ref-payment-batching/venv/lib/python3.4/site-packages/bitcoin/rpc.py", line 428, in getblock
    return CBlock.deserialize(unhexlify(r))
  File "/home/harding/ref-payment-batching/venv/lib/python3.4/site-packages/bitcoin/core/serialize.py", line 115, in deserialize
    r = cls.stream_deserialize(fd)
  File "/home/harding/ref-payment-batching/venv/lib/python3.4/site-packages/bitcoin/core/__init__.py", line 523, in stream_deserialize
    vtx = VectorSerializer.stream_deserialize(CTransaction, f)
  File "/home/harding/ref-payment-batching/venv/lib/python3.4/site-packages/bitcoin/core/serialize.py", line 255, in stream_deserialize
    r.append(inner_cls.stream_deserialize(f))
  File "/home/harding/ref-payment-batching/venv/lib/python3.4/site-packages/bitcoin/core/__init__.py", line 329, in stream_deserialize
    vin = VectorSerializer.stream_deserialize(CTxIn, f)
  File "/home/harding/ref-payment-batching/venv/lib/python3.4/site-packages/bitcoin/core/serialize.py", line 255, in stream_deserialize
    r.append(inner_cls.stream_deserialize(f))
  File "/home/harding/ref-payment-batching/venv/lib/python3.4/site-packages/bitcoin/core/__init__.py", line 199, in stream_deserialize
    scriptSig = script.CScript(BytesSerializer.stream_deserialize(f))
  File "/home/harding/ref-payment-batching/venv/lib/python3.4/site-packages/bitcoin/core/serialize.py", line 234, in stream_deserialize
    return ser_read(f, l)
  File "/home/harding/ref-payment-batching/venv/lib/python3.4/site-packages/bitcoin/core/serialize.py", line 78, in ser_read
    raise SerializationError('Asked to read 0x%x bytes; MAX_SIZE exceeded' % n)
bitcoin.core.serialize.SerializationError: Asked to read 0x4c8f597226cf3302 bytes; MAX_SIZE exceeded
```

**Expected results**

The second print_block() should be similar to the first.

**Other information**

Block 482439 used above isn't even over the old consensus limit:

```text
bitcoin-cli getblock $( bitcoin-cli getblockhash 482439 ) | jq '[.size, .strippedsize, .weight]'
[
  999257,
  999221,
  3996920
]
```

I've tried a few other blocks going back to the segwit activation and they also generate the same error even though they too were below the old consensus limit.

**Comment by petertodd** [comment]
That's to be expected: python-bitcoinlib-v0.8.0 doesn't have segwit support. See the release notes: https://github.com/petertodd/python-bitcoinlib/blob/master/release-notes.md#v080

As mentioned in those release notes, you can use the Bitcoin Core `-rpcserialversion=0` option to disable segwit in the RPC interface, which should fix the problem (if it doesn't, let me know!).

segwit support is in master, but I haven't had time to review that code well enough to feel confident releasing it; it will be released as v0.9.0 as it's API will be not-fully-backwards compatible with v0.8.0. Though I expect the breakage between v0.9.0 and v0.8.0 to be less than between v0.7.x and v0.8.0

**Comment by harding** [comment]
Sorry.  I somehow read those notes the first time as saying segwit support was provided, although on a re-read it's very clear that it's not.  Thanks!
